Reworked ATR Tilt Speed conditions

NicoAleman opened the pull request:
Fix: Reworked ATR Tilt Speed conditions:
 - Re-organized implementation to avoid repeated logic
 - Accounts for ATR Thresholds in the Boost margin
 - Uses "On" Tilt Speed for Transition Boost if faster

The margins originally in place to determine whether or not to use Transition Boost were before ATR Thresholds, when ATR Setpoint would constantly jump around between positive and negative when cruising. With these thresholds in place, the margins are less necessary and achieve the same effect. This fix negates the thresholds from the margins in effect, making the response the same as originally intended (unless the thresholds are larger than the original margins, then those are respected).

Additionally, depending on configuration and your current speed, there can be situations where Tilt Speed * the current Response Boost is faster than the Transition Boost speed. So for transition boost, we look at both tilt speeds and use the fastest one available, similarly to how transitions are now handled in Torque Tiltback in my other Pull Request.

The logic was also just reorganized as a whole, the original structure had unnecessary repeating logic and was slightly hard to follow. It's still a lot here thanks to the new logic, but I consolidated what I can.

NicoAleman committed:
ATR: Refactor implementation

No behavior changes, just re-structuring in preparation for further updates.

lukash commented:
@NicoAleman it seems your commit should be split into three commits, according to the bullet points in your commit message. Can you please do that? That way it's way easier to understand the changes. I also came to look at the simplification you did, as I'm making other changes and I thought I'd pull that in right away to minimize conflicts, but like this I can't do it.

(Also the code cleanup changes don't need to go to the changelog)

NicoAleman committed:
Workflow: Update CI clang-tools reference

NicoAleman committed:
ATR: Rework Tilt Speed Conditions

Unify the tilt speed selection between Uphill and Downhill. Rather than Down -> Up boosting with Transition Boost, and Up -> Down "boosting" with On_Step_Size, both states use the faster of the two tilt speeds.

The ERPM Threshold unique to Down -> Up was also deemed unnecessary and removed.

Fix: Rework ATR to more effectively boost tilt speeds in fast transitions

NicoAleman committed:
ATR: Account for ATR Thresholds in Boost margin

Fix: High ATR Thresholds no longer make Transition Boost harder to trigger

NicoAleman commented:
@lukash Let me know if there's any issue. Had to update the CI file to pass checks, not sure if appropriate to include in this PR but wasn't sure how best to go about it otherwise.

lukash commented:
This has also been reworked for 1.3 and the changes are no longer relevant, closing...
